src/types.ts

```diff
@@ -261,3 +261,73 @@ export interface CreatedComment {
 export const CreatedCommentSpec: ObjectSpec<CreatedComment> = {
   id: int,
 };
+
+export interface SetFlag {
```

Comment:
I was trying to utilize the existing `Flag` interface using `Omit<Flag, ...>`, but I wasn't able to figure out how to apply omit on `FlagSpec`.

Reply:
There are a few options here. I think the simplest and cleanest is probably just to flip things around, define `Flag` in terms of `SetFlag` instead of vice-versa:
```javascript
export interface SetFlag {
  name: string;
  type_id: int;
  status: string;
  requestee: string | undefined;
}
export const SetFlagSpec: ObjectSpec<SetFlag> = {
  name: string,
  type_id: int,
  status: string,
  requestee: optional(string),
};

export type Flag = SetFlag & {
  id: int;
  creation_date: datetime;
  modification_date: datetime;
  setter: string;
}
export const FlagSpec: ObjectSpec<Flag> = {
  ...SetFlagSpec,
  id: int,
  creation_date: datetime,
  modification_date: datetime,
  setter: string,
};
```

Reply:
That is great point, thanks, I'll do that. :+1: